instrumentation/que/lib/opentelemetry/instrumentation/que/patches/poller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Que
+      module Patches
+        # Instrumentation for the Que::Poller module
+        module Poller
+          def poll(*)
+            # Avoid tracing when should_poll? returns true. This is also used
+            # in Poller#poll to decide if the actuall poll should be executed
+            # or not. Without this we would generate a lot of unnecessary
+            # spans.
+            return unless should_poll?
```

Comment:
I am concerned of the risk here that this will possibly end up changing the implementation/behavior of the target class `Que::Poll#poll`. Is there a way to instrument this class in a way that leaves the behavior in tact and rely on the Gem implementation without reimplementing the logic in the instrumentation?

Reply:
I also wanted to avoid changing this but I didn't find any other way. The [poll](https://github.com/que-rb/que/blob/61e1805deb8161e5c1502e518a9a44fa464d308c/lib/que/poller.rb#L140-L173) function does the SQL query directly in the same function, and without the should_poll guard we'd get way too many spans.

Reply:
Maybe manage the spans manually instead of the `in_span` helper and checking for a `nil` return value from super to see if the instrumentation should record a span?

Reply:
Hmm, interesting idea. I think this could work. The only downside I see is that we'd create a lot of spans that we would discard immediately. About 20 per second. Do you think that would be okay or do you think this is overhead we cannot afford?

Reply:
It would not be ideal to
generate unused objects. There also may be cases where pg spans end up missing parents and that would be extremely unpleasant for our users so my suggestion is not helpful.

This makes me want to reach out to the maintainers and get their feedback. I wonder if they would be willing to add or change their internal logger hooks and allow us to leverage that for instrumentation:

https://github.com/que-rb/que/blob/master/lib/que/utils/logging.rb#L35

Is that something you'd be willing to try?

Reply:
I think the existing maintainer is not currently maintaining the project. There's an [Issue open](https://github.com/que-rb/que/issues/313) about adding new maintainers which currently seems to be in progress. I think after that there's a likelihood of getting new improvements/changes into the que repository.